core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala

```diff
@@ -0,0 +1,238 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.{Actor, Props}
+import akka.pattern.{after, pipe}
+import cats._
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, MethodRepoDAO, SlickDataSource, UserDirectoryDAO}
+import org.broadinstitute.dsde.rawls.google.GooglePubSubDAO
+import org.broadinstitute.dsde.rawls.model.Subsystems._
+import org.broadinstitute.dsde.rawls.model.{StatusCheckResponse, SubsystemStatus}
+import org.broadinstitute.dsde.rawls.monitor.HealthMonitor._
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+/**
+  * Created by rtitle on 5/17/17.
+  */
+object HealthMonitor {
+  val DefaultFutureTimeout = 1 minute
+  val DefaultStaleThreshold = 15 minutes
+
+  val OkStatus = SubsystemStatus(true, None)
+  val UnknownStatus = SubsystemStatus(false, Some(List("Unknown status")))
+  def failedStatus(message: String) = SubsystemStatus(false, Some(List(message)))
+
+  // Actor API:
+
+  /** Triggers subsystem checking */
+  case object CheckAll
```

Comment:
We usually make the case classes/objects we use for Actor message extend a common base class per service... but having looked into it, I have no idea why. We don't use the base class anywhere.